main_server/local_server/server_admin/server_admin_javascript.js

```diff
@@ -4822,31 +4846,93 @@ function BMLT_Server_Admin()
                 };
             };
         };
-                
+
         if ( !key_match ) {
             formatType_menu.selectedIndex = 0;
         };
-        
+
         formatType_td.appendChild(formatType_menu);
-    
+
     /************************************************************************************//**
     *   \brief  Handle data input from the text items.                                      *
     ****************************************************************************************/
         this.handleFormatTypeSelectInput = function ( in_select_input_object ///< This will contain the affected select input
                                             ) {
             var new_value = in_select_input_object.options[in_select_input_object.selectedIndex].value;
-        
+
             for (var c = 0; c < g_langs.length; c++) {
                 in_select_input_object.format_group_objects[g_langs[c]].type = new_value;
             };
-        
+
             this.evaluateFormatState(in_select_input_object.shared_id);
         };
         /***************************************
          * End Format-Type-Enum
          ******************************************/
         /**********************************************/
- 
+
+        this.getFormatForMasterId = function(format_code, language) {
+            for (var index = 0; index < g_formats_array.length; index++) {
+                var format_group = g_formats_array[index];
+                if (format_group['formats']['en']['key'] === format_code) {
+                    return format_group['formats'][language]
+                }
+            }
+        }
+
+        this.isVenueTypeFormatByCode = function(format_code) {
+            return this.venueTypeFormats[format_code] !== undefined;
+        }
+
+        this.isVenueTypeFormatBySharedId = function(shared_id) {
+            for (var i = 0; i < this.venueTypeFormatSharedIds.length; i++) {
+                if (this.venueTypeFormatSharedIds[i] === shared_id) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        this.venueTypeFormatSharedIds = [
+            parseInt(this.getFormatForMasterId('VM', g_langs[0])['shared_id']),
```

Comment:
If it turns out that someone out there has done something silly and deleted one of these formats, I don't mind adding a database migration to add it back. After it's added back, because we're disabling deletion of these going forward, we should be set.

Reply:
I think there was another ticket addressing this right?  https://github.com/bmlt-enabled/bmlt-root-server/issues/362

Reply:
Yes, but that doesn't prevent someone from deleting them before they upgrade. I'm saying it's worth checking to see what folks have done, and I might need to write a migration to be doubly sure they exist depending on what we find.

Reply:
Ok I'm not sure how to handle this.